src/Core/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -55,12 +56,16 @@ public class CosmosSqlMetadataProvider : ISqlMetadataProvider
 
         public CosmosSqlMetadataProvider(RuntimeConfigProvider runtimeConfigProvider, IFileSystem fileSystem)
         {
+            RuntimeConfig runtimeConfig = runtimeConfigProvider.GetConfig();
             _fileSystem = fileSystem;
-            _runtimeConfig = runtimeConfigProvider.GetConfig();
+            // Many classes have references to the RuntimeConfig, therefore to guarantee
+            // that the Runtime Entities are not mutated by another class we make a copy of them
+            // to store internally.
+            _runtimeConfigEntities = new RuntimeEntities(runtimeConfig.Entities.Entities);
```

Comment:
are we doing this because we dont want the entities or isDevelopmentMode to be refreshed for CosmosSqlMetadataProvider?